### PR TITLE
[CI] Added support for building rust-binding on MacOS.

### PR DESCRIPTION
**Pull request by apepkuss**
This PR is to add support for building rust-binding on MacOS.

- Fixed #1269

### CODE REVIEW

File: .github/workflows/bindings-rust.yml [hunk 1]
```diff
@@ -9,72 +9,147 @@ on:
...
+          cargo test --lib --examples --locked
+          cargo test --doc --locked
+
+  build_macos:
+    name: MacOS
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [macos-11]
+    needs: get_version_v2
```
Comment (hydai) [inline .github/workflows/bindings-rust.yml:92]: Not sure why do you need this?

**Review by hydai — CHANGES_REQUESTED** [review]

File: .github/workflows/bindings-rust.yml (hunk 1 above)
Comment (apepkuss) [reply .github/workflows/bindings-rust.yml:92]: According to the description from [GitHub](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners), `macos-latest` or `macos-11` denotes `macOS Big Sur 11` currently. When a new version of MacOS is released in the future, for example, macos-12, `macos-latest` cannot cover macos-11 at that time.

Comment (hydai) [reply .github/workflows/bindings-rust.yml:92]: I mean the `get_version_v2`. This is useless.

Comment (apepkuss) [reply .github/workflows/bindings-rust.yml:92]: By using [macos-11] (or [macos-11, macos-11] in the future),  I think we can have a much finer-grained control on the OS version. Thanks!

Comment (hydai) [reply .github/workflows/bindings-rust.yml:92]: I am fine with the `macos-11` part.
However, looks like there is a misunderstanding. What does this line `needs: get_version_v2` mean?

Comment (apepkuss) [reply .github/workflows/bindings-rust.yml:92]: > 

Sorry, I thought the comment was for line 91. I'll remove it, thanks a lot!

**Comment by apepkuss** [comment]
Hi @hydai, the issue has been fixed, could you please help review the latest change? Thanks a lot!

**Comment by hydai** [comment]
@apepkuss 
The macos-11 is still failed. Is this an expected result?

**Comment by apepkuss** [comment]
@hydai I'll check the issue, thanks!

**Comment by apepkuss** [comment]
@hydai The issues related to `binding-rust/MacOS` have been fixed. But there is still a failure. According to my investigation, the failure happened in the `build_windows` job defined in `build.yml`. This PR does not involve this file, so I'm not sure if this is an expected result or not.

**Comment by hydai** [comment]
The test should pass. However, it's very strange. I will take a look.

**Comment by hydai** [comment]
@apepkuss 
Looks like the environment changed. I have created an new issue #1290 to trace this.
And I am going to merge this PR first

**Comment by apepkuss** [comment]
@hydai Thanks for your help and reminder!

**Review by hydai — APPROVED** [review]

**Pull request merged**